### PR TITLE
Add enumerated types

### PR DESCRIPTION
**Pull request by godcrampy**
in reference to #61

### CODE REVIEW

**Comment by godcrampy** [comment]
@googlebot  I signed it!

**Comment by kevmoo** [comment]
Hrm. I guess this is tighter, more idiomatic Dart. Thoughts, @natebosch ?

**Comment by kevmoo** [comment]
Ooops! I missed that one:
https://pub.dev/documentation/yaml/latest/yaml/CollectionStyle-class.html

File: lib/src/style.dart [hunk 1]
```diff
@@ -46,27 +46,18 @@ class ScalarStyle {
 }
 
 /// An enum of collection styles.
-class CollectionStyle {
-  /// No source style was specified.
-  ///
+enum CollectionStyle {
```
[the remark this answers is not in the document]
Comment (godcrampy) [reply lib/src/style.dart]: Made requested changes.

**Comment by natebosch** [comment]
Thanks!

**Pull request merged**